src/middlewared/middlewared/alert/source/certificates.py

```diff
@@ -59,84 +39,118 @@ class CertificateRevokedAlertClass(AlertClass):
+                'type': _type,
+            },
+            {
+                'id': (await self.middleware.call('webdav.config'))['certssl'],
+                'service': 'Webdav',
+                'type': _type,
+            },
+            {
+                'id': (await self.middleware.call('openvpn.server.config'))['server_certificate'],
+                'service': 'OpenVPN Server',
+                'type': _type,
+            },
+            {
+                'id': (await self.middleware.call('openvpn.client.config'))['client_certificate'],
+                'service': 'OpenVPN Client',
+                'type': _type,
+            },
+            {
+                'id': (await self.middleware.call('system.general.config'))['ui_certificate']['id'],
+                'service': 'Web UI',
+                'type': _type,
+            },
+            {
+                'id': (await self.middleware.call('system.advanced.config'))['syslog_tls_certificate'],
+                'service': 'Syslog',
+                'type': _type,
+            },
+        ]
+        return service_certs
+
+    async def _get_cas(self):
+        _type = 'root certificate authority'
+        ca_certs = [
+            {
+                'id': (await self.middleware.call('openvpn.server.config'))['root_ca'],
+                'service': 'Web UI',
+                'type': _type,
+            },
+            {
+                'id': (await self.middleware.call('openvpn.client.config'))['root_ca'],
+                'service': 'Syslog',
+                'type': _type,
+            },
+        ]
+        return ca_certs
 
-class CertificateParsingFailedAlertSource(AlertSource):
     async def check(self):
         alerts = []
+        certs = await self.middleware.call('certificate.query', [['certificate', '!=', None]])
+        certs.extend(await self.middleware.call('certificateauthority.query'))
 
-        for cert in await self.middleware.call(
-                'certificate.query',
-                [['certificate', '!=', None]]
-        ) + await self.middleware.call('certificateauthority.query'):
+        # make the sure certs have been parsed correctly
+        parsed = []
+        for cert in certs:
             if not cert['parsed']:
-                alerts.append(
-                    Alert(
-                        CertificateParsingFailedAlertClass,
-                        {
-                            "type": cert["cert_type"].capitalize(),
-                            "name": cert["name"],
-                        },
-                    )
-                )
+                alerts.append(Alert(
+                    CertificateParsingFailedAlertClass,
+                    {"type": cert["cert_type"].capitalize(), "name": cert["name"]},
+                ))
+            else:
+                parsed.append(cert)
+
+        if parsed:
+            revoked_certs = [i for i in await self._get_service_certs() + await self._get_cas() if i['id']]
```

Comment:
It would be nice if we could reword the variable name please as it seems from the variable that these certs have been revoked which is not the case and we can potentially make a mistake for future edits